Feature/#5 post

HTH016 opened the pull request:
### JWT 
- Claim에 회원 id를 추가했습니다.

### Post
- Post 작성 및 조회 기능을 구현했습니다.

HTH016 committed:
#05
jwt 토큰 claim에 memberId 추가
포스트 작성, 포스트 목록 조회, 포스트 조회 기능 추가

HTH016 committed:
#05
jwt 토큰 claim에 memberId 추가
포스트 작성, 포스트 목록 조회, 포스트 조회 기능 추가

HTH016 committed:
#05 수정
jwt 설정 클래스 코드 간소화

HTH016 committed:
Merge branch 'feature/#3-category' into feature/#5-post

# Conflicts:
#	src/main/java/flab/Linkedlog/config/JwtProperties.java
#	src/main/java/flab/Linkedlog/repository/MemberRepository.java
#	src/main/java/flab/Linkedlog/util/JwtUtil.java
#	src/test/java/flab/Linkedlog/MemberServiceTests.java

HTH016 committed:
#05 (수정)
테스트 코드 추가
ApiResponse 컨벤션 변경

HTH016 committed:
#05 (수정)
테스트 코드 추가
ApiResponse 컨벤션 변경
dto 이름 변경

HTH016 committed:
Merge branch 'feature/#3-category' into feature/#5-post

HTH016 committed:
#05 (수정)
post 테스트 코드 작성

HTH016 committed:
Merge branch 'feature/#5-post' of https://github.com/f-lab-edu/linked-log into feature/#5-post

mason1155 commented on src/main/java/flab/Linkedlog/repository/PostRepositoryObsolete.java:
```diff
@@ -0,0 +1,44 @@
+import java.util.List;
+
+@Repository
+public interface PostRepositoryObsolete extends JpaRepository<Post, Long> {
```
이 파일은 안쓰는 것으로 보이는데 제거해도 되지 않을까요?

mason1155 commented on src/main/java/flab/Linkedlog/repository/post/PostRepositoryCustomImpl.java:
```diff
@@ -0,0 +1,74 @@
+    @Transactional
+    public void incrementViewCount(Long postId) {
+        // EntityManager로 Post 객체 조회
+        Post post = entityManager.find(Post.class, postId);
```
entityManager를 직접 쓰기 보단 PostRepository로 조회하고 update하는 것을 권장드립니다..

mason1155 commented on src/main/java/flab/Linkedlog/service/PostService.java:
```diff
@@ -0,0 +1,117 @@
+            throw new IllegalArgumentException("The post does not exist in the category");
+        }
+
+        postRepository.incrementViewCount(postId);
```
여기서 post를 가지고 update하기만 해도 됩니다.

mason1155 commented on src/test/java/flab/Linkedlog/postTest/PostDetailTests.java:
```diff
@@ -0,0 +1,275 @@
+
+        assertThat(result.getResponse().getStatus()).isEqualTo(HttpStatus.BAD_REQUEST.value());
+
+        ApiResponse<PostDetailResponse> apiResponse = objectMapper.readValue(responseContent, new TypeReference<>() {
```
불필요한 코드입니다.

mason1155 commented on src/test/java/flab/Linkedlog/postTest/CreatePostTests.java:
```diff
@@ -0,0 +1,197 @@
+        String jsonResponse = result.getResponse().getContentAsString();
+        ApiResponse response = objectMapper.readValue(jsonResponse, ApiResponse.class);
+
+        assertThat(result.getResponse().getStatus()).isEqualTo(200);
```
응답 코드는 andExpect로 검증해도 되지 않을까요?

mason1155 commented on src/main/java/flab/Linkedlog/service/PostService.java:
```diff
@@ -0,0 +1,117 @@
+    @Transactional(readOnly = true)
+    public List<PostListResponse> searchPostsByCategoryAndKeyword(Long categoryId, String keyword) {
+
+        if (keyword == null || keyword.trim().isEmpty()) {
```
이런 검증은 컨트롤러에서 해도 되지 않을까요?

mason1155 commented on src/main/java/flab/Linkedlog/service/PostService.java:
```diff
@@ -0,0 +1,117 @@
+    }
+
+    @Transactional
+    public Optional<PostDetailResponse> getPostDetailById(Long categoryId, Long postId) {
```
Optional에 감싸서 전달할 필요는 없어보여요

HTH016 committed:
#05 (수정)
test code status 검증 변경
조회수 증가 로직 변경
Postservice 예외 처리 위치, return 타입 변경
QueryDslConfig 추가

mason1155 commented on src/main/java/flab/Linkedlog/controller/PostController.java:
```diff
@@ -0,0 +1,72 @@
+    @PreAuthorize("permitAll()")
+    public ApiResponse<List<PostListResponse>> searchPostsByCategoryAndKeyword(
+            @PathVariable Long categoryId,
+            @RequestParam String keyword) {
```
페이징도 적용해주세요

mason1155 commented on src/main/java/flab/Linkedlog/dto/post/PostDetailResponse.java:
```diff
@@ -0,0 +1,43 @@
+public class PostDetailResponse {
+
+
+    @NotEmpty(message = "글 아이디")
```
Long에는 NotEmpty가 적용 안되요

mason1155 commented on src/main/java/flab/Linkedlog/dto/post/PostDetailResponse.java:
```diff
@@ -0,0 +1,43 @@
+    private String content;
+
+    @NotEmpty(message = "작성일")
+    private LocalDateTime createdAt;
```
LocalDateTime도 NotEmpty가 적용 안되요

HTH016 committed:
#05 수정
dto validation 어노테이션 수정
pagination 추가
빈 목록에 대한 테스트 추가

mason1155 reviewed (approved):
고생하셨습니다
